src/odemis/gui/util/widgets.py

```diff
@@ -180,8 +180,12 @@ def _on_value_change(self, evt):
         """ This method is called when the value of the control is changed.
         it moves the axis to the new value.
         """
+        evt.Skip()
+
         try:
             if self._future is not None and not self._future.done():
+                # Note: when the value is truncated, COMMAND_ENTER is triggered
+                # twice, which is one of the reasons this path is taken.
                 logging.info("Not moving axis %s as it's already moving", self.axis)
```

Comment:
Actually, I meant "clipped" instead of "truncated". For instance, if you enter "1m" in the wavelength widget, it'll be clipped to the maximum possible wavelength.

Reply:
I'll push a fixed version.